sqlglot/dialects/clickhouse.py

```diff
@@ -261,6 +387,14 @@ def _parse_join(
                 join.set("global", join.args.pop("method", None))
             return join
 
+        def _get_combinator_parts(self, func_name: str) -> t.List[str]:
```

Comment:
this is extremely inefficient. can you instead leverage trie.py? build the trie as a class constant and use it here to do prefix matching. you can reverse the string and feed it into the trie

Reply:
Yeah, I can use trie, didn't know that it's there.

Reply:
Maybe it's better to put all agg func names as prefixes in a trie then?
Or both.

Reply:
sure, do whatever is cleanest / most efficient

Reply:
i guess to avoid reverse of string, you can put the inverse in the trie

Reply:
I've tried a quite a lot of approaches, and it looks like trie is not faster...
I.e. it's faster for shorter functions, like:
```python
_agg_func_trie = new_trie(AGG_FUNCTIONS)
print(timeit(lambda: in_trie(_agg_func_trie, "quantile"), number=10000))
```
```
0.003980431938543916
```
```python
timeit(lambda: _get_combinator_parts("quantile"), number=10000)
```
```
0.007004864048212767
```
But for longer ones it's exactly the same or slower.
And unfortunately one search in a trie doesn't cut it, as I still need to know where the suffix ends.
Trie that counts depth is immediately slower...

Reply:
Just adding all strings to a huge dict is 10x faster. lol
```
0.0008398909121751785
```

Reply:
perfect use a dict